tools/oldest_constraints.txt

```diff
@@ -50,12 +50,6 @@ ConfigArgParse==0.10.0
 funcsigs==0.4
 zope.hookable==4.0.4
 
-# Ubuntu Bionic constraints
-# Formerly only lexicon==2.2.1 is available on Bionic. But we cannot put that here because some
-# DNS plugins require higher versions. We put to the least minimal Lexicon version to ensure
-# that Lexicon 2.x works with Certbot.
-dns-lexicon==2.7.14
```

Comment:
Yes, you are right. Better to stick to Lexicon 2.x for oldest, and Lexicon 3.x for pinned versions + dev.

Reply:
I wasn't very clear here. I think we should put `dns-lexicon==2.2.1` in `oldest_constraints.txt`, remove that version from the relevant requirements file, and pin a newer version where we need to such as the OVH plugin. 2.2.1 is the constraint from Ubuntu and I think we should try and make all of our Lexicon plugins work with this version of Ubuntu unless there's a reason we're unable to for a specific plugin.

Reply:
I am hesitating here. Yes formely it should be 2.2.1, because of Bionic. But the particular value of Lexicon does not make much sense anyway, because only one provider is supported on Ubuntu Bionic. For the other providers, the oldest requirements test is in fact useless.

So we are free to upgrade to any version of Lexicon we want for the other providers in fact. And so we need to scatter specific requirements on every local-oldest-requirements.txt. No the best maintainability in my opinion.

What make sense here, is to have a Lexicon 2.x version, to ensure that Certbot run on both branches 2.x and 3.x.

So for that matter, why not put here the latest 2.x version, with the relevant comment to explain the intention, and set only a specific 2.2.1 constraint on dnsimple, for this only provider that can be installed on Bionic?

Reply:
The reason I think it's valuable to default to keeping support with the version of Lexicon in Ubuntu Bionic is it gives us the option of adding other DNS plugins that work with that version of Lexicon to the OS repos in the future. It's true that we've had very limited success making things like that happen and we're probably not going that direction with packaging, but we have general approval from Ubuntu to do things like that and I think it costs us very little to keep compatibility where possible until we've decided on an approach for distributing Certbot.

Reply:
You mean, in theory we are authorized to submit SRUs with new providers on a LTS release?

Reply:
Yes. Someone just has to do the work to build the .debs for them to review.

Reply:
Then if it is possible, indeed appropriate strategy is to use oldest as its initial purpose.